lib/rex/socket/ssl.rb

```diff
@@ -13,10 +13,12 @@ module Rex::Socket::Ssl
 
   module CertProvider
 
-    def self.ssl_generate_subject(cn = Rex::Text.rand_hostname, org = Rex::Text.rand_name)
-      st  = Rex::Text.rand_state
-      loc = Rex::Text.rand_name.capitalize
-      "/C=US/ST=#{st}/L=#{loc}/O=#{org.capitalize}/CN=#{cn}"
+    def self.ssl_generate_subject(cn: nil, org: nil, loc: nil, st: nil)
+      st  ||= Rex::Text.rand_state
+      loc ||= Rex::Text.rand_name.capitalize
+      org ||= Rex::Text.rand_name.capitalize
+      cn  ||= Rex::Text.rand_hostname
+      "/C=US/ST=#{st}/L=#{loc}/O=#{org}/CN=#{cn}"
```

Comment:
At first I thought "Hmm, why not expand this to country", but figured that then having to deal with state/country mismatch wouldn't be fun.

Reply:
Yeah, I'd like to have less duplication between this and https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/cert_provider.rb overall, but I figured I'd leave it at getting them to equivalent-ish interfaces for now.